### PR TITLE
Update readme with some mention of new architecture

### PR DESCRIPTION
**Pull request by chrisglein**
### Type of Change
Documentation only. No product changes.

### Why
Readme didn't cover some FAQs around new architecture, and was missing key links to helper repos.

### What
- Add summary of Fabric roadmap with link to pinned issue
- Move Requirements section to be a subsection of "Getting Started"
- Move "Documentation" before "Contributing"
- Flesh out "Examples" section with missing samples repo and gallery repo references
- Tighten up various language

## Changelog
Should this change be included in the release notes: No
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13192)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,44 +28,43 @@ Visit the official [React Native for Windows + macOS website](https://microsoft.
...
+### Logging Issues
+Search the [existing issues](https://github.com/microsoft/react-native-windows/issues) and try to make sure your problem doesn’t already exist before opening a new issue. If your issue doesn't exist yet, provide as much information as possible so we can better help you. Include the information requested by the [appropriate issue template](https://github.com/microsoft/react-native-windows/issues/new/choose).
 
 ## Documentation
-[React Native already has great documentation](https://reactnative.dev/docs/getting-started) and we're working to ensure the React Native Windows is part of that documentation story.
-
-[React Native for Windows](https://microsoft.github.io/react-native-windows/) has it's own separate documentation site where Windows and macOS specific information, like API docs and blog updates live.
+React Native has [great documentation](https://reactnative.dev/docs/getting-started). React Native for Windows adds it's own separate [Windows and macOS documentation](https://microsoft.github.io/react-native-windows/) for desktop platform information like API docs and blog updates.
```
Comment (stmoy) [inline README.md:56]:
```suggestion
React Native has [great documentation](https://reactnative.dev/docs/getting-started). React Native for Windows adds its own separate [Windows and macOS documentation](https://microsoft.github.io/react-native-windows/) for desktop platform information like API docs and blog updates.
```

File: README.md [hunk 2]
```diff
@@ -28,44 +28,43 @@ Visit the official [React Native for Windows + macOS website](https://microsoft.
...
 
 ### Examples
 - Using the CLI in the [Getting Started](https://microsoft.github.io/react-native-windows/docs/getting-started) guide will set you up with a sample React Native for Windows app that you can begin editing right away.
-- If you're looking for sample code, just browse the RNTester folder in the GitHub web UI
+- Check the [samples repo](https://github.com/microsoft/react-native-windows-samples) for more standalone samples.
+- The [React Native Gallery](https://github.com/microsoft/react-native-gallery) app demonstrates various components in an interactive way.
+- For more sample code browse the [RNTester folder](https://github.com/microsoft/react-native-windows/tree/main/packages/e2e-test-app/windows/RNTesterApp) in the GitHub web UI
```
Comment (stmoy) [inline README.md:62]: Consider adding a link to the devblog:

```
- Check out the [React Native Developer Blog](https://devblogs.microsoft.com/react-native/) to see examples from past conference talks, blog posts, and more.
```

**Review by stmoy — APPROVED** [review]
Looks good! Added some minor feedback.

File: README.md [hunk 3]
```diff
@@ -28,44 +28,43 @@ Visit the official [React Native for Windows + macOS website](https://microsoft.
...
-
-[React Native for Windows](https://microsoft.github.io/react-native-windows/) has it's own separate documentation site where Windows and macOS specific information, like API docs and blog updates live.
+React Native has [great documentation](https://reactnative.dev/docs/getting-started). React Native for Windows adds it's own separate [Windows and macOS documentation](https://microsoft.github.io/react-native-windows/) for desktop platform information like API docs and blog updates.
 
 ### Examples
 - Using the CLI in the [Getting Started](https://microsoft.github.io/react-native-windows/docs/getting-started) guide will set you up with a sample React Native for Windows app that you can begin editing right away.
-- If you're looking for sample code, just browse the RNTester folder in the GitHub web UI
+- Check the [samples repo](https://github.com/microsoft/react-native-windows-samples) for more standalone samples.
+- The [React Native Gallery](https://github.com/microsoft/react-native-gallery) app demonstrates various components in an interactive way.
```
Comment (chrisglein) [inline README.md:61]:
```suggestion
- The [React Native Gallery](https://github.com/microsoft/react-native-gallery) app demonstrates various components in an interactive way.
- Check out the [React Native Developer Blog](https://devblogs.microsoft.com/react-native/) to see examples from past conference talks, blog posts, and more.
```

**Review by marlenecota — APPROVED** [review]

**Pull request merged**